components/kyma-environment-broker/internal/process/provisioning/event_hub_step.go

```diff
@@ -140,6 +147,39 @@ func (p *ProvisionAzureEventHubStep) Run(operation internal.ProvisioningOperatio
 	return operation, 0, nil
 }
 
+// getUniqueNSName will try to use suffix to generate a unique name for Event Hubs namespace, or generate a random one
+// if it already exists
+func (p *ProvisionAzureEventHubStep) getUniqueNSName(suffix string, azureClient azure.AzureInterface,
+	log logrus.FieldLogger) (string, error) {
+	uniqueName := fmt.Sprintf("skr-%s", suffix)
+	u, err := azureClient.CheckNamespaceAvailability(p.context, uniqueName)
+	if err != nil {
+		msg := "Error while calling Azure Event Hubs client. Can't check namespace name availability. Error: %v"
+		log.Errorf(msg, err)
+		return "", errors.Errorf(msg, err)
+	}
+
+	for i := retryCount; !u && i > 0; i-- {
+		newName := fmt.Sprintf("skr-%s", uuid.New().String())
+		log.Info("Namespace %s already exists. Checking new name %s.", uniqueName, newName)
+		uniqueName = newName
+		u, err = azureClient.CheckNamespaceAvailability(p.context, uniqueName)
+		if err != nil {
+			msg := "Error while calling Azure Event Hubs client. Can't check namespace name availability. Error: %v"
+			log.Errorf(msg, err)
+			return "", errors.Errorf(msg, err)
+		}
+	}
+
+	if !u {
+		msg := fmt.Sprintf("Can't find a unique name for Event Hubs namespace. Failed after %d attempts.",
+			retryCount)
+		log.Error(msg)
+		return "", errors.New(msg)
+	}
+	return uniqueName, nil
+}
+
```

Comment:
(nit) Calls to CheckNamespaceAvailability and error creation could be moved to an inner function to avoid some repetition and improved readability. 
Please use whatever version you prefer. Just an idea :)

```suggestion
func (p *ProvisionAzureEventHubStep) getUniqueNSName(suffix string, azureClient azure.AzureInterface,
	log logrus.FieldLogger) (string, error) {
	checkAvailability := func(uniqueName string) (bool, error) {
		isUnique, err := azureClient.CheckNamespaceAvailability(p.context, uniqueName)
		if err != nil {
			msg := "Error while calling Azure Event Hubs client. Can't check namespace name availability. Error: %v"
			log.Errorf(msg, err)
			return false, errors.Errorf(msg, err)
		}
		return isUnique, nil
	}

	uniqueName := fmt.Sprintf("skr-%s", suffix)
	isUnique, err := checkAvailability(uniqueName)
	if err != nil {
		return "", err
	}

	for i := retryCount; !isUnique && i > 0; i-- {
		newName := fmt.Sprintf("skr-%s", uuid.New().String())
		log.Info("Namespace %s already exists. Checking new name %s.", uniqueName, newName)
		uniqueName = newName
		isUnique, err = checkAvailability(uniqueName)
		if err != nil {
			return "", err
		}
	}

	if !isUnique {
		msg := fmt.Sprintf("Can't find a unique name for Event Hubs namespace. Failed after %d attempts.",
			retryCount)
		log.Error(msg)
		return "", errors.New(msg)
	}
	return uniqueName, nil
}
```

Reply:
(nit) Isn't this a perfect fit for a unit test? 😎

Reply:
> (nit) Calls to CheckNamespaceAvailability and error creation could be moved to an inner function to avoid some repetition and improved readability.

Or we get rid off the 2 different ways to generate names (as humans cannot use them anyway) and reduce to just one function ... no need for a closure here